hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -92,36 +102,69 @@ class IncrementalRelation(val sqlContext: SQLContext,
   override def schema: StructType = latestSchema
 
   override def buildScan(): RDD[Row] = {
-    val fileIdToFullPath = mutable.HashMap[String, String]()
+    val regularFileIdToFullPath = mutable.HashMap[String, String]()
+    var metaBootstrapFileIdToFullPath = mutable.HashMap[String, String]()
+
     for (commit <- commitsToReturn) {
       val metadata: HoodieCommitMetadata = HoodieCommitMetadata.fromBytes(commitTimeline.getInstantDetails(commit)
         .get, classOf[HoodieCommitMetadata])
-      fileIdToFullPath ++= metadata.getFileIdAndFullPaths(basePath).toMap
+
+      if (HoodieTimeline.METADATA_BOOTSTRAP_INSTANT_TS == commit.getTimestamp) {
```

Comment:
bootstrap is one time operation done during migration. not sure if the user wants to incrementally fetch that existing data as a single batch. Not sure what it means to incrementally query from bootstrap instant ts?  can we just avoid this whole thing by only supporting incremental query on commit timestamps after bootstrap?

Reply:
I had discussed this with Balaji, and we concluded to invest in this so that there is no difference in terms of user experience w.r.t whether the table is bootstrapped or not. Right now users can query data in the first commit written through incremental query, and we wanted to support this experience with bootstrapped tables as well.

A use-case I can think of:

User needs to access data until a provided `end commit` (from `first commit` till some provided `end commit`). This is possible to do with incremental query in regular Hudi tables. We would want to support this experience even with bootstrapped tables.

Reply:
@bvaradar can chime in here as well, so we can reach a conclusion

Reply:
Hmmm. that seems fair. although I would have probably done this after an ask came up. Since you have it already, let's just keep this.